### PR TITLE
Add `debug pieces` subcommand

### PR DESCRIPTION
**Pull request by danieldk**
This subcommand prints piece statistics for a given model and train/dev corpora.

Example output:

```
================== Training corpus statistics ===================
Median token length: 1.0
Mean token length: 1.34
Token length range: [1, 24]

================== Development corpus statistics ================
Median token length: 1.0
Mean token length: 1.32
Token length range: [1, 24]
```

## Checklist

- [ ] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [ ] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: curated_transformers/cli/debug_pieces.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from spacy.cli._util import (
+    debug_cli,
+    import_code,
+    parse_config_overrides,
+    show_validation_error,
+)
+from spacy.schemas import ConfigSchemaTraining
+from spacy.tokens import Doc
+from spacy.util import registry, resolve_dot_names
+from typer import Argument as Arg, Context, Option as Opt
```
Comment (svlandeg) [inline curated_transformers/cli/debug_pieces.py:14]: Note that we'll be [moving](https://github.com/explosion/spaCy/pull/12258) from `typer` to `radicli` for `v4` so maybe it makes sense already using that here now?

File: curated_transformers/cli/debug_pieces.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    # fmt: off
+    ctx: Context,  # This is only used to read additional arguments
+    config_path: Path = Arg(..., help="Path to config file", exists=True, allow_dash=True),
+    code_path: Optional[Path] = Opt(None, "--code-path", "--code", "-c", help="Path to Python file with additional code (registered functions) to be imported"),
+    transformer_name: Optional[str] = Opt(None, "--name", "-n", help="Name of the transformer pipe to gather piece statistics for (default: first transformer pipe)."),
+    # fmt: on
+):
+    """
+    Analyze word- or sentencepiece statistics.
+    DOCS: https://spacy.io/api/cli#debug-data
```
Comment (svlandeg) [inline curated_transformers/cli/debug_pieces.py:35]:
```suggestion
    DOCS: https://spacy.io/api/cli#debug-pieces
```
(and TODO: actually write those docs :-) )

File: curated_transformers/cli/debug_pieces.py (hunk 1 above)
Comment (danieldk) [reply curated_transformers/cli/debug_pieces.py:14]: But this needs to be compatible with spaCy 3.6, would combining these work? We plan to make a `v4` branch of curated transformers for distillation. Maybe we can change it there later?

Comment (svlandeg) [reply curated_transformers/cli/debug_pieces.py:14]: It probably wouldn't work if spaCy 3.6 were used with an older Python version, so yea, let's keep this for the `v4` branch.

**Pull request merged**